legend-engine-xts-graphQL/legend-engine-xt-graphQL-grammar/src/test/java/org/finos/legend/engine/language/graphQL/grammar/test/roundtrip/TestGraphQLRoundtrip.java

```diff
@@ -147,6 +147,46 @@ public void testQueryRoundtrip()
                 "}");
     }
 
+    @Test
+    public void testQueryWithDirectiveRoundtrip()
+    {
+        check("query getUserWithProjects($a: INT = 1) @cool {\n" +
+                "  user(id: 2) @totalCount @directiveWithParams(param1: \"dummy\", param2: \"dummy value\") {\n" +
+                "    firstname(x: null, x: $ok)\n" +
+                "    lastname(test: true)\n" +
+                "    projects(other: \"oo\") {\n" +
+                "      name\n" +
+                "      tasks(val: en) {\n" +
+                "        description(p: 3.2, z: [1, 2, \"ok\"])\n" +
+                "      }\n" +
+                "    }\n" +
+                "  }\n" +
+                "}");
+    }
+
+    @Test
+    public void testDirectiveParsingError()
```

Comment:
This is a test which should be written in the parser tests and not roundtrip tests

Reply:
We don't have parser tests in GraphQL module. And we have other error tests in roundtrip. It can be done in a separate changeset.